ipykernel/debugger.py

```diff
@@ -434,17 +434,23 @@ async def source(self, message):
             'request_seq': message['seq'],
             'command': message['command']
         }
-        source_path = message["arguments"]["source"]["path"]
-        if os.path.isfile(source_path):
-            with open(source_path, encoding='utf-8') as f:
-                reply['success'] = True
-                reply['body'] = {
-                    'content': f.read()
-                }
+        source_ref = message["arguments"].get("sourceReference", 0)
+        if source_ref == 0:
+            message["arguments"].get("source", {}).get("sourceReference", 0)
```

Comment:
```suggestion
            source_ref = message["arguments"].get("source", {}).get("sourceReference", 0)
```

Reply:
@r3m0t Thanks for testing. This change is meant to solve this for non-vscode usages as well 👍